pkg/tokenvalidation/expirationvalidator.go

```diff
@@ -0,0 +1,31 @@
+package oauth
+
+import (
+	"errors"
+	"time"
+
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	userv1 "github.com/openshift/api/user/v1"
+)
+
+var errExpired = errors.New("token is expired")
+
+func NewExpirationValidator() OAuthTokenValidator {
+	return OAuthTokenValidatorFunc(
+		func(token *oauthv1.OAuthAccessToken, _ *userv1.User) error {
+			if token.ExpiresIn > 0 {
```

Comment:
is this a hard limit after which the token expires?

Reply:
Yes, the next two lines test it